src/vm/method.hpp

```diff
@@ -3360,7 +3363,7 @@ class InstantiatedMethodDesc : public MethodDesc
         //
         // For generic method definitions that are not the typical method definition (e.g. C<int>.m<U>)
         // this field is null; to obtain the instantiation use LoadMethodInstantiation
-    PTR_Dictionary m_pPerInstInfo;  //SHARED
+    RelativePointer<PTR_Dictionary> m_pPerInstInfo;  //SHARED
```

Comment:
https://github.com/dotnet/coreclr/issues/11363#issuecomment-299081077 has discussion about field not accessed from JITed code.

This field is accessed from JITed code. I am not sure whether it is a good idea to do unconditionally on all platforms. It may need more scrutiny. Maybe best to separate the change for this one from the change for other fields so that it is easier to make forward progress on this change.